plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -743,19 +759,17 @@ enterComputeCek = computeCek
     returnCek (FrameApplyFun fun ctx) arg =
         applyEvaluate ctx fun arg
     -- s , [_ V1 .. Vn] ◅ lam x (M,ρ)  ↦  s , [_ V2 .. Vn]; ρ [ x  ↦  V1 ] ▻ M
-    returnCek (FrameApplyValues args ctx) fun = case args of
-        (arg:rest) -> applyEvaluate (FrameApplyValues rest ctx) fun arg
```

Comment:
Note that you kinda process `args` twice here yourself. One matching per argument to get to the frame + one matching to get to the actual argument (at which point you reconstruct the frame).

Reply:
That's a good point: we still allocate N `FrameApplyValues` constructors overall!